### DISCUSSION TITLE
Is there a standalone version of ESLint for a browser extension?

### DISCUSSION THREAD

**Discussion by erosman** [eslint/eslint#17179] [Q&A]
Is there a standalone ESLint version to use for an editor (CodeMirror 5) in a Firefox extension?

**Comment by Rec0iL99** [comment]
Hi @erosman, if you are looking to use ESLint in a browser environment, you might want to look into the [`Linter`](https://eslint.org/docs/latest/integrate/nodejs-api#linter) class, this is the class that gets called for linting. We do a similar thing in the [ESLint playground](https://github.com/eslint/eslint.org/blob/a1462b7f2c170f90419a9af7e2a8f751af5f97b7/src/playground/App.js#L8).

**Comment by erosman** [comment]
Which files have to be included to run Linter class?
The example uses node.js  `import` from `'./node_modules/eslint/lib/linter/'`

**Comment by kaizhu256** [comment]
- likely majority of eslint files looking @ [eslint/lib/linter/linter.js](https://github.com/eslint/eslint/blob/main/lib/linter/linter.js#L14)

- the closest thing to a "standalone version of ESLint for a browser" appears to be the rollup file
https://eslint.org/assets/js/playground.js (2mb minified)

    - i would probably hack the rollup to expose the `Linter()` class to window/global scope, and then call the class from codemirror.

**Comment by erosman** [comment]
[playground.js](https://eslint.org/assets/js/playground.js) seems rather large and includes React which is not needed.

**Reply by Rec0iL99** [reply]
We don't have a proper distribution for the `Linter` class in a build, so you would have to understand the playground source code a bit to make it work. I found this [npm package](https://www.npmjs.com/package/eslint-linter-browserify) today that might help you here.
